docs:#1552 doc controller snapshots (#1658)
redpanda-data/documentation: modules/manage/pages/schema-reg/schema-reg-api.adoc

When you register an evolved schema for an existing subject, the version `id` is incremented by 1.

== Use Schema Registry contexts

Starting in Redpanda v26.1, you can use contexts to create isolated namespaces for schemas, subjects, and configuration within a single Schema Registry instance.

ifndef::env-cloud[]

> To use contexts, first set the `schema_registry_enable_qualified_subjects` cluster configuration property to `true`, then restart the broker. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for setup instructions.

Reviewer: ```suggestion
To use contexts, first set the `schema_registry_enable_qualified_subjects` cluster configuration property to `true`, then restart the brokers. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for setup instructions.
```

== Use Schema Registry contexts

Starting in Redpanda v26.1, you can use contexts to create isolated namespaces for schemas, subjects, and configuration within a single Schema Registry instance.

ifndef::env-cloud[]
To use contexts, first set the `schema_registry_enable_qualified_subjects` cluster configuration property to `true`, then restart the broker. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for setup instructions.
endif::env-cloud[]
ifdef::env-cloud[]
To use contexts on BYOC and Dedicated clusters, xref:manage:cluster-maintenance/config-cluster.adoc[configure the cluster property] `schema_registry_enable_qualified_subjects`. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for details.
endif::env-cloud[]

Contexts are created implicitly when you register a schema using a context-qualified subject. For example, registering a schema with the subject `:.staging:sensor-value` creates the `.staging` context if it does not already exist:

[source,bash]
----
curl -s -X POST \
  http://localhost:8081/subjects/:.staging:sensor-value/versions \
  -H "Content-Type: application/vnd.schemaregistry.v1+json" \
  -d '{"schema": "{\"type\":\"string\"}"}'
----

> 

Reviewer: ```suggestion
```

When you register an evolved schema for an existing subject, the version `id` is incremented by 1.

== Use Schema Registry contexts

> Starting in Redpanda version 26.1, you can use contexts to create isolated namespaces for schemas, subjects, and configuration within a single Schema Registry instance.

Reviewer: ```suggestion
Starting in Redpanda v26.1, you can use contexts to create isolated namespaces for schemas, subjects, and configuration within a single Schema Registry instance.
```

ifndef::env-cloud[]
To use contexts, first set the `schema_registry_enable_qualified_subjects` cluster configuration property to `true`, then restart the broker. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for setup instructions.
endif::env-cloud[]
ifdef::env-cloud[]
To use contexts on BYOC and Dedicated clusters, xref:manage:cluster-maintenance/config-cluster.adoc[configure the cluster property] `schema_registry_enable_qualified_subjects`. See xref:manage:schema-reg/schema-reg-contexts.adoc[] for details.
endif::env-cloud[]

Contexts are created implicitly when you register a schema using a context-qualified subject. For example, registering a schema with the subject `:.staging:sensor-value` creates the `.staging` context if it does not already exist:

[source,bash]
----
curl -s -X POST \
  http://localhost:8081/subjects/:.staging:sensor-value/versions \
  -H "Content-Type: application/vnd.schemaregistry.v1+json" \
  -d '{"schema": "{\"type\":\"string\"}"}'
----

=== Scope API operations to a context

> 

Reviewer: ```suggestion

```

You can scope any Schema Registry API operation to a specific context in two ways:

* Qualified subjects: Use the `:<context>:<subject>` syntax wherever a subject is accepted. For example, to list subjects in the `.staging` context:
+
[source,bash]
----
curl -s "http://localhost:8081/subjects?subjectPrefix=:.staging:"
----

* Base URL prefix: Prefix any endpoint path with `/contexts/{context}/`. The server rewrites the request to operate within that context. This approach is recommended for non-Java SerDe clients that do not support qualified subject syntax:
+
[source,bash]
----
curl -s http://localhost:8081/contexts/.staging/subjects

> ----

Reviewer 2: This is still WIP, as per https://github.com/redpanda-data/docs/pull/1658/changes#r3123466186

Reviewer: So remove the entire point about the base URL prefix?

Reviewer 2: @trevpanda @mattschumpert what do you think?

We need to either remove it or call out that it's not yet supported. But from slack I vaguely recall a policy that we don't document future features, so I am leaning removing this for now, yes.

Reviewer 3: Yes, we should remove until we support

> ----

Reviewer: ```suggestion
```

Unqualified subjects continue to work and operate in the default context.

> For the complete endpoint reference, see the link:/api/doc/schema-registry/[Schema Registry API reference].

Reviewer: ```suggestion

```
